subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -244,9 +261,28 @@ public void setupProjectLayout() {
             throw new GradleException("Package name is not supported for '" + initDescriptor.getId() + "' build type.");
         }
 
+        InsecureProtocolOption insecureProtocolOption = null;
+        if (initDescriptor instanceof PomProjectInitDescriptor) {
+            final PomProjectInitDescriptor pomProjectInitDescriptor = (PomProjectInitDescriptor) initDescriptor;
+
+            if (!isNullOrEmpty(this.insecureProtocolOption)) {
+                insecureProtocolOption = InsecureProtocolOption.byId(this.insecureProtocolOption);
+                if (insecureProtocolOption == null) {
+                    final TreeFormatter formatter = new TreeFormatter();
+                    formatter.node("The requested insecure protocol handler '" + getInsecureProtocolHandler() + "' is unknown. Supported options");
+                    formatter.startChildren();
+                    pomProjectInitDescriptor.getInsecureProtocolHandlers().forEach(h -> formatter.node("'" + h.getId() + "'"));
+                    formatter.endChildren();
+                    throw new GradleException(formatter.toString());
+                }
+            } else {
+                insecureProtocolOption = pomProjectInitDescriptor.getDefaultInsecureProtocolHandler();
```

Comment:
I think we can get rid of `getDefaultInsecureProtocolHandler` if we're using a property and rely on `insecureProtocolOption` always having a value